### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by yehorGuyAvrahamCom**

### CODE REVIEW

File: src/js/App.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import theme from './Theme/Theme';
+
+
+
+export function App(): JSX.Element {
+
+    return (
+      <ChakraProvider theme = {theme}>
+          <GlobalSettingsProvider>
+              <MainContainer/>
```
Comment (yehorGuyAvrahamCom) [inline src/js/App.tsx:16]: Any code formatting: tab size 2 and 4, trailing comma exist and doesn't exist, excess empty lines

File: src/js/GlobalSettings/GlobalSettingsProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    React.useEffect(() => {
+        
+        hardware = detectHardware();
+        reloadElement();
+    }, []);
+
+
+    return (
+        <GlobalSettingsContext.Provider value = {{hardware, isPhoneHardware, mainContent, setMainContent, connectWay}}>
```
Comment (yehorGuyAvrahamCom) [inline src/js/GlobalSettings/GlobalSettingsProvider.tsx]: Use memorized object for context value

File: src/js/GlobalSettings/Hardware/Hardware.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+export const HARDWARE_TYPES: Hardware[] = ['iphone', 'android', 'windows', 'mac'];
+
+
+
+export function isPhoneHardware(hardware: Hardware): boolean {
+    
+    return (hardware === 'android' || hardware === 'iphone');
+}
+
+export function detectHardware(): Hardware {
```
Comment (yehorGuyAvrahamCom) [inline src/js/GlobalSettings/Hardware/Hardware.tsx:13]: better to use the special lib for this

File: src/js/GlobalSettings/useGlobalSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import { GlobalSettingsContext } from './GlobalSettingsProvider';
+
+
+
+export default function useGlobalSettings() {
+
+    const globalSettings = React.useContext(GlobalSettingsContext)
```
Comment (yehorGuyAvrahamCom) [inline src/js/GlobalSettings/useGlobalSettings.tsx]: This variable can be `undefined`, this case be handled

File: src/js/MainContainer/MainContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                <Transactions/>
+            }
+        </Flex>
+    </Flex>
+    );
+}
+
+
+
+export default React.memo(MainContainer);
```
Comment (yehorGuyAvrahamCom) [inline src/js/MainContainer/MainContainer.tsx]: The `React.memo` is an expensive operation; better don't use it without particular reason

File: src/js/GlobalSettings/GlobalSettingsProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+export default function GlobalSettingsProvider({children}: IProviderProps): JSX.Element {
+
+    const {mainContent, setMainContent} = useMainContent();
+    const {reloadElement} = useReloadElement();
+    const connectWay = React.useRef<ConnectWay>('');
+
+    React.useEffect(() => {
+        
+        hardware = detectHardware();
```
Comment (yehorGuyAvrahamCom) [inline src/js/GlobalSettings/GlobalSettingsProvider.tsx]: Why do we need a global variable? 
Better replace it to `const hardware = useMemo(() => detectHardware(), []);`

File: src/js/components/AlertDialogErrorProvider/AlertDialogErrorProvider.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+export const AlertDialogContext = React.createContext<IAlertDialogErrorProvider>(DEFAULT_ALERT_DIALOG_CONTEXT_VALUE);
+
+
+
+function AlertDialogError({children}: IProviderProps): JSX.Element {
+
+    const [isOpen, setIsOpen] = React.useState<boolean>(false);
+    const closeButtonRef = React.useRef<any>();
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/AlertDialogErrorProvider/AlertDialogErrorProvider.tsx]: Don't use `any` type

File: src/js/components/ConnectMetamask/MetamaskProvider.tsx [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+
+
+
+const DEFAULT_ACCOUNT_VALUE: IAccount = {
+    account: null,
+    chainId: null,
+    balance: null,
+    icon: null
+}
+
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/ConnectMetamask/MetamaskProvider.tsx]: Repeated object

File: src/js/components/ConnectMetamask/MetamaskProvider.tsx [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+        const isProviderDetected = await detectProvider();
+
+        callback(isProviderDetected);
+    }, [providerState]);
+
+    const updateAccountData = (accountName: string) => {
+
+        let acc: IAccount = {...account, account: accountName} 
+        
+        providerState!.request({method: 'eth_getBalance', params: [accountName, 'latest']})
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/ConnectMetamask/MetamaskProvider.tsx]: Don't use the `!` operator

File: src/js/components/Header/AccountMenu/AccountMenu.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+                        onClick = {() => setIsConnectModalOpen(previous => previous = true)}
+                        fontSize = {'sm'}
+                    >
+                        Connect Wallet
+                    </Button>
+                }
+            </Flex>
+            <Modal
+                isOpen = {(isAccountModalOpen && account.account || isConnectModalOpen) ? true : false}
+                onClose = {() => { setIsAccountModalOpen(previous => previous = false); setIsConnectModalOpen(previous => previous = false) }}
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/Header/AccountMenu/AccountMenu.tsx]: Need to create named variable and function

File: src/js/components/Header/AccountMenu/ModalContents/AccountContent.tsx [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+                        flex = {1}
+                    >
+                        <Text opacity={0.5} fontSize = {'sm'}>Connected with Metamask</Text>
+                        <Spacer/>
+                        <Button
+                            size = {isPhoneHardware(hardware) ? 'lg': 'xs'}
+                            borderColor={'red'}
+                            sx = {smallButtonStyle}
+                            onClick = {() => {onDisconnect(); setIsAccountModalOpen(previous => previous = false);}}
+                        >
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/Header/AccountMenu/ModalContents/AccountContent.tsx]: Named handler

File: src/js/components/Header/AccountMenu/ModalContents/AccountContent.tsx [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+                            Disconnect
+                        </Button>
+                        <Button
+                            size = {isPhoneHardware(hardware) ? 'lg': 'xs'}
+                            sx = {smallButtonStyle}
+                            borderColor={'blue'}
+                            onClick = {() => {setIsAccountModalOpen(previous => previous = false); setIsConnectModalOpen(previous => previous = true)}}
+                        >
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/Header/AccountMenu/ModalContents/AccountContent.tsx]: The same

File: src/js/components/Header/AccountMenu/ModalContents/ConnectWalletContent.tsx [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+        });
+    }
+
+    const turnOffConnecting = () => {
+
+        if(isConnecting === false) return;
+
+        setIsConnecting(previous => previous = false);
+        setIsAccountModalOpen(previous => previous = true);
+        setIsConnectModalOpen(previous => previous = false);
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/Header/AccountMenu/ModalContents/ConnectWalletContent.tsx]: What is it? Should it be comparing like `...(previous => previous === false)`? 
Or is it just `setIs.....(false)`?

File: src/js/components/Header/DropdownMenu/DropdownMenu.tsx [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+
+    const isTransactionsDisabled: boolean = account.account 
+                                            ? 
+                                            (
+                                                (account.chainId === NETWORKS_BY_NAME['Ropsten'] || 
+                                                account.chainId === NETWORKS_BY_NAME['Goerli'] || 
+                                                account.chainId === NETWORKS_BY_NAME['Ethereum'] || 
+                                                account.chainId === NETWORKS_BY_NAME['Rinkeby'] || 
+                                                account.chainId === NETWORKS_BY_NAME['Kovan'])
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/Header/DropdownMenu/DropdownMenu.tsx]: `!Object.values(NETWORKS_BY_NAME).includes(account.chainId)`

File: src/js/utils/Hooks/convertToHex.tsx [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+    '11': 'b',
+    '12': 'c',
+    '13': 'd',
+    '14': 'e',
+    '15': 'f'
+}
+
+
+
+export default function convertToHex(number: number) {
```
Comment (yehorGuyAvrahamCom) [inline src/js/utils/Hooks/convertToHex.tsx:1]: `"0x" + Number(number).toString(16)` doesn't work correctly?

File: src/js/components/AlertDialogErrorProvider/AlertDialogErrorProvider.tsx (hunk 7 above)
Comment (GuyAvraham) [reply src/js/components/AlertDialogErrorProvider/AlertDialogErrorProvider.tsx]: @NikitaGuyAvrahamCom also add eslint here with rule that will avoid any type
@yehorGuyAvrahamCom can show u where we have it ready. talk to him on slack

**Comment by GuyAvraham** [comment]
please also add eslint with prettier and all what we have in pastime
go to slack and ask help from Yehor

File: src/js/GlobalSettings/GlobalSettingsProvider.tsx (hunk 2 above)
Comment (NikitaGuyAvrahamCom) [reply src/js/GlobalSettings/GlobalSettingsProvider.tsx]: Do you mean, I should to create an object that will contain all context values and put it into useMemo?

File: src/js/GlobalSettings/GlobalSettingsProvider.tsx [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+}: IProviderProps): JSX.Element {
+  const { mainContent, setMainContent } = useMainContent();
+  const connectWay = React.useRef<ConnectWay>("");
+
+  const hardware: Hardware = React.useMemo(() => detectHardware(), []);
+
+  const contextValue = React.useMemo(() => {
+    return {
+      hardware,
+      isPhoneHardware,
```
Comment (yehorGuyAvrahamCom) [inline src/js/GlobalSettings/GlobalSettingsProvider.tsx]: Arrow functions have the shortcut view for return value
```
  const contextValue = React.useMemo(() => ({
    hardware,
    ...
```

File: src/js/GlobalSettings/Hardware/Hardware.tsx [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+export const HARDWARE_TYPES: Hardware[] = [
+  "iphone",
+  "android",
+  "windows",
+  "mac",
+];
+
+export function isPhoneHardware(): boolean {
+  return isMobile;
+}
```
Comment (yehorGuyAvrahamCom) [inline src/js/GlobalSettings/Hardware/Hardware.tsx]: I think better to delete it and just use the value from lib)

File: src/js/GlobalSettings/useGlobalSettings.tsx [hunk 18]
```diff
@@ -0,0 +1,13 @@
+import * as React from "react";
+import { GlobalSettingsContext } from "./GlobalSettingsProvider";
+
+export default function useGlobalSettings() {
+  const globalSettings = React.useContext(GlobalSettingsContext);
+
+  if (globalSettings === undefined)
+    throw new Error(
```
Comment (yehorGuyAvrahamCom) [inline src/js/GlobalSettings/useGlobalSettings.tsx]: 1. the condition can be replaced to `if (!globalSettings)`
2. You added throw of errors to the code, but do you have the place where you are handling it?

Comment (yehorGuyAvrahamCom) [reply src/js/GlobalSettings/useGlobalSettings.tsx]: https://ru.react.js.org/docs/error-boundaries.html

File: src/js/components/ConnectMetamask/MetamaskProvider.tsx [hunk 19]
```diff
@@ -0,0 +1,241 @@
...
+      await providerState.on("accountsChanged", onAccountChanged);
+      await providerState.on("connect", onConnect);
+      await providerState.on("disconnect", onDisconnect);
+    };
+
+    const onProviderChanged = async () => {
+      if (!providerState)
+        throw new Error("providerState equals to " + providerState);
+
+      await subscribeEvents();
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/ConnectMetamask/MetamaskProvider.tsx:201]: add the catch block for handle an errors what you throw :)

File: src/js/components/ConnectMetamask/MetamaskProvider.tsx [hunk 20]
```diff
@@ -0,0 +1,241 @@
...
+            alertDialogError(
+              "Error connecting",
+              "The connection attempt failed. Please try to connect in your wallet.",
+              "Try again"
+            );
+          }
+        });
+    };
+
+    onProviderChanged();
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/ConnectMetamask/MetamaskProvider.tsx:223]: the same

File: src/js/components/Header/AccountMenu/AccountMenu.tsx [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+    setIsConnectModalOpen(false);
+  };
+
+  const onAccountModalOpen = () => {
+    setIsAccountModalOpen(true);
+  };
+
+  const onConnectModalOpen = () => {
+    setIsConnectModalOpen(true);
+  };
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/Header/AccountMenu/AccountMenu.tsx]: For these cases Chakra-UI has nice hook `useDisclosure`
https://chakra-ui.com/docs/hooks/use-disclosure

File: src/js/components/Header/Dapp/useConnector.tsx [hunk 22]
```diff
@@ -0,0 +1,123 @@
...
+export default function useConnector(): UseConnectorReturnValue {
+  const [connector, setConnector] = React.useState<WalletConnect | null>(
+    DEFAULT_CONNECTOR_VALUE
+  );
+  const { reloadElement } = useReloadElement();
+  const { alertDialogError } = useAlertDialogError();
+
+  React.useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let connector: any = localStorage.getItem("walletconnect");
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/Header/Dapp/useConnector.tsx]: it will be string, not any 🤪

File: src/js/components/Transactions/Transactions.tsx [hunk 23]
```diff
@@ -0,0 +1,154 @@
...
+  return new Date(timeStamp * 1000).toString();
+}
+
+function ethParse(value: number) {
+  return value / 10 ** 18;
+}
+
+function Transactions(): JSX.Element {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [transactions, setTransactions] = React.useState<any[] | null>(null);
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/Transactions/Transactions.tsx]: create an interface for Transactions

Comment (NikitaGuyAvrahamCom) [reply src/js/components/Transactions/Transactions.tsx]: Done

File: src/js/components/Header/Dapp/useConnector.tsx (hunk 22 above)
Comment (NikitaGuyAvrahamCom) [reply src/js/components/Header/Dapp/useConnector.tsx]: Done

File: src/js/components/Header/AccountMenu/AccountMenu.tsx (hunk 21 above)
Comment (NikitaGuyAvrahamCom) [reply src/js/components/Header/AccountMenu/AccountMenu.tsx]: Done

File: src/js/components/ConnectMetamask/MetamaskProvider.tsx (hunk 20 above)
Comment (NikitaGuyAvrahamCom) [reply src/js/components/ConnectMetamask/MetamaskProvider.tsx:223]: Done

File: src/js/components/ConnectMetamask/MetamaskProvider.tsx (hunk 19 above)
Comment (NikitaGuyAvrahamCom) [reply src/js/components/ConnectMetamask/MetamaskProvider.tsx:201]: Done

File: src/js/GlobalSettings/useGlobalSettings.tsx (hunk 18 above)
Comment (NikitaGuyAvrahamCom) [reply src/js/GlobalSettings/useGlobalSettings.tsx]: Done

File: src/js/GlobalSettings/Hardware/Hardware.tsx (hunk 17 above)
Comment (NikitaGuyAvrahamCom) [reply src/js/GlobalSettings/Hardware/Hardware.tsx]: Done

File: src/js/GlobalSettings/GlobalSettingsProvider.tsx (hunk 16 above)
Comment (NikitaGuyAvrahamCom) [reply src/js/GlobalSettings/GlobalSettingsProvider.tsx]: Done

File: src/js/components/Wallet/Balance/NewBlockEvent/NewBlockEvent.tsx [hunk 24]
```diff
@@ -0,0 +1,80 @@
...
+        newBlockInfo.current.filterId = filterId;
+
+        setUpdater((previous) => !previous);
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }, [account.account]);
+
+  return <></>;
```
Comment (yehorGuyAvrahamCom) [inline src/js/components/Wallet/Balance/NewBlockEvent/NewBlockEvent.tsx]: You can replace this component to custom hook or replace return to `null`

Comment (NikitaGuyAvrahamCom) [reply src/js/components/Wallet/Balance/NewBlockEvent/NewBlockEvent.tsx]: Done!

**Pull request merged**